### PR TITLE
OBPIH-2663 Unit price of product supplier record should update upon p…

### PR DESCRIPTION
**Pull request by jmiranda**
…lace PO

### CODE REVIEW

File: grails-app/services/org/pih/warehouse/core/UomService.groovy [hunk 1]
```diff
@@ -21,6 +21,7 @@ class UomService {
             eq("product", product)
             eq("uom", unitOfMeasure)
             eq("quantity", quantity)
+            isNull("prodcut")
```
Comment (jmiranda) [inline grails-app/services/org/pih/warehouse/core/UomService.groovy:24]: Ah, this is wrong.

File: grails-app/services/org/pih/warehouse/order/OrderStatusEventService.groovy [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+    OrderService orderService
+    UomService uomService
+
+    void onApplicationEvent(OrderStatusEvent event) {
+        log.info "Application event ${event} has been published!"
+        OrderStatus orderStatus = event?.source
+        Order order = Order.get(event?.order?.id)
+
+        log.info "Order ${order.orderNumber} ${order.status} has just transitioned to status ${orderStatus}"
```
Comment (jmiranda) [inline grails-app/services/org/pih/warehouse/order/OrderStatusEventService.groovy:32]: Remove logging as well as next line.

File: grails-app/views/product/_productSuppliers.gsp [hunk 3]
```diff
@@ -55,14 +53,18 @@
...
-                            <td>${fieldValue(bean: productSupplier, field: "unitOfMeasure")}</td>
 
                             <td>
-                                <g:hasRoleFinance onAccessDenied="${g.message(code:'errors.blurred.message', args: ['0.00'])}">
-                                    ${fieldValue(bean: productSupplier, field: "unitPrice")}
-                                    ${grailsApplication.config.openboxes.locale.defaultCurrencyCode}
-                                </g:hasRoleFinance>
+                                ${productSupplier.productPackages.size()}
                             </td>
+%{--                            <td>${fieldValue(bean: productSupplier, field: "unitOfMeasure")}</td>--}%
```
Comment (jmiranda) [inline grails-app/views/product/_productSuppliers.gsp:60]: Need to revisit what we want to do here.

**Comment by jmiranda** [comment]
@awalkowiak @DominikaHatala Can you guys review this today?

**Comment by awalkowiak** [comment]
@jmiranda sure, doing it RN

**Pull request merged**